server/routes/offer.routes.js

```diff
@@ -0,0 +1,162 @@
+import express from 'express';
+import { authenticate, schemaValidation, isAdmin, hasValidObjectId } from '../helpers/middleware';
+import { createOfferSchema, acceptOfferSchema, assignOfferSchema } from '../schemas/offer.schema';
+import OfferController from '../controllers/offer.controllers';
+
+const router = express.Router();
+
+/**
+ * @swagger
+ * /offer/create:
+ *   post:
+ *     tags:
+ *       - Offer
+ *     description: Creates a new offer
+ *     produces:
+ *       - application/json
+ *     requestBody:
+ *      content:
+ *        application/json:
+ *          schema:
+ *            $ref: '#/components/schemas/Offer'
+ *      description: Add a new offer
+ *     responses:
+ *      '201':
+ *        description: Offer created
+ *      '400':
+ *        description: Error creating offer
+ *      '500':
+ *       description: Internal server error
+ */
+router.post(
+  '/create',
+  authenticate,
+  isAdmin,
+  schemaValidation(createOfferSchema),
+  OfferController.create,
+);
+
+/**
+ * @swagger
+ * /offer/accept:
+ *   put:
+ *     tags:
+ *       - Offer
+ *     description: Accepts a specific offer
+ *     produces:
+ *       - application/json
+ *     requestBody:
+ *      content:
+ *        application/json:
+ *          schema:
+ *            $ref: '#/components/schemas/Offer'
+ *      description: Respond to a specific offer
+ *     responses:
+ *      '200':
+ *        description: Offer accepted
+ *      '400':
+ *        description: Error accepting to offer
+ *      '500':
+ *       description: Internal server error
+ */
+router.put('/accept', authenticate, schemaValidation(acceptOfferSchema), OfferController.accept);
+
+/**
+ * @swagger
+ * /offer/assign:
+ *   put:
+ *     tags:
+ *       - Offer
+ *     description: Assigns a specific offer to a user
+ *     produces:
+ *       - application/json
+ *     requestBody:
+ *      content:
+ *        application/json:
+ *          schema:
+ *            $ref: '#/components/schemas/Offer'
+ *      description: Assigns a specific offer to a user
+ *     responses:
+ *      '200':
+ *        description: Offer assigned
+ *      '400':
+ *        description: Error assigning offer
+ *      '500':
+ *       description: Internal server error
+ */
+router.put(
+  '/assign',
+  authenticate,
+  isAdmin,
+  schemaValidation(assignOfferSchema),
+  OfferController.assign,
+);
+
+/**
+ * @swagger
+ * /offer/all:
+ *   get:
+ *     tags:
+ *       - Offer
+ *     description: Get all offers
+ *     produces:
+ *       - application/json
+ *     requestBody:
+ *      content:
+ *        application/json:
+ *          schema:
+ *            $ref: '#/components/schemas/Offer'
+ *      description: Get all owned offers
+ *     responses:
+ *      '200':
+ *        description: returns object of offers
+ *      '500':
+ *       description: Internal server error
+ */
+router.get('/all', authenticate, OfferController.getAll);
```

Comment:
It looks like this gets all the offers for a single user. Can we update the route so that it is more explicit.

```suggestion
router.get('/user/all', authenticate, OfferController.getAll);
```

Can we also create an equivalent route for an admin?

Reply:
this gets all the offers for a user based on their `userId`.